### ISSUE TITLE
Grey objects with no textures rendering

### ISSUE THREAD

**Issue by bit-garden** [zesage/flutter_cube#36]
Looks like the texture isn't loading but the meshes are still able to render.

Heres what I see with the cube example.
![image](https://user-images.githubusercontent.com/10319571/134724035-d43f8d2f-62d1-41c3-a5c0-b0e8b02cb008.png)

There was a prior issue that mentioned this, but it was self closed with no posted solution.

**Comment by will-thomson4** [comment]
I came up against this issue. I was able to fix it by firstly by making sure all my files (.obj, .mtl, .png) were in one folder and was added to the pubspec.yaml like this.

`assets: `
   `- assets/Orangutan/`
   
However, more importantly, I think this issue is due to the .obj file not being able to find the textures associated with it. In order to have a fully textured model you need the .obj file itself which has the model geometry, but also a .mtl file which has information about the material and textures associated with that model. That might look something like this:

#Blender MTL File: 'Orangutan.blend'
#Material Count: 1
newmtl Material
Ns 323.999994
Ka 1.000000 1.000000 1.000000
Kd 0.800000 0.800000 0.800000
Ks 0.500000 0.500000 0.500000
Ke 0.000000 0.000000 0.000000
Ni 1.450000
d 1.000000
illum 2
map_Kd color.png

And then finally you need the textures themselves which are usually in .png or .jpg format, these are referenced in the .mtl file. As you can see at the bottom of the example I posted above, color.png references a png file called color which is inside the same directory, and it needs to be written as it is there with `map_Kd FILENAME`.

I found that not all of the .obj files I was downloading from the internet also included a .mtl which correctly referenced the textures for the model, so I had to manually edit some of the .mtl files to correctly include the path to the textures.

TLDR: Put .png, .mtl, .obj files in same directory and make sure .mtl file references the correct path to the image texture files.

**Comment by praveenjaiswal16** [comment]
loadTexture(material, basePath, isAsset: isAsset) change in library

**Comment by ext-danny** [comment]
Is it possible to use only .obj and .mtl files without the png?

**Comment by will-thomson4** [comment]
> Is it possible to use only .obj and .mtl files without the png?

I'm not an expert on this, but I believe if you do this you will get the result above where you have object geometry but just white surfaces, as the .png files have the images which are mapped onto the surface as the textures/colors.
